src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -3598,4 +3599,106 @@ void M(file x) { }
         }
         EOF();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75992")]
+    public void TestFileModifierAfterIncompleteBaseList1()
+    {
+        UsingTree("""
+            class C : B
+            file class D
```

Comment:
I imagine a similar, more contrived case may also be positively impacted

```cs
class Outer
{
    class C : B

    required int Prop { get; set; }
}
```

but, don't find it interesting enough to ask you to come back and add the test.